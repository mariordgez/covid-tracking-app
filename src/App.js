import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import NavBar from './components/NavBar';
import Countries from './components/Countries';
import Country from './components/Country';

function App() {
  return (
    <Router>
      <NavBar />
      <Switch>
        <Route exact path="/">
          <Countries />
        </Route>
        <Route path="/country/:id">
          <Country />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
