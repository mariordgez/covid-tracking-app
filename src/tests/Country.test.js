import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import App from '../App';

describe('Rockets component', () => {
  describe('Use Jest snapshots to test the Country component', () => {
    it('renders Country component as expected', async () => {
      const CountryComponent = render(
        <Provider store={store}>
          <App />
        </Provider>,
      );

      expect(CountryComponent.asFragment()).toMatchSnapshot();
    });
  });
});
