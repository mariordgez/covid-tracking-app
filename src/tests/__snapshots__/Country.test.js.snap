// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component Use Jest snapshots to test the Country component renders Country component as expected 1`] = `
<DocumentFragment>
  <nav
    class="bg-blue-light navbar navbar-expand-lg navbar-light"
  >
    <div
      class="container"
    >
      <div
        class="me-auto d-flex justify-content-between flex-row w-100 navbar-nav"
      >
        <div
          class="nav-item"
        >
          <a
            aria-current="page"
            class="fw-bold m-3 text-decoration-none text-white active"
            href="/"
          >
            Home
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            aria-current="page"
            class="fw-bold m-3 text-decoration-none text-white active"
            href="/"
          >
            Countries
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="pt-3 bg-blue-home"
  >
    <div
      class="m-0 row"
    >
      <div
        class="d-flex justify-content-end col-md-6 col-sm-6 col-6"
      >
        <img
          alt="World"
          height="150px"
          src="earth.svg"
        />
      </div>
      <div
        class="p-0 text-white d-flex flex-column justify-content-center col-md-6 col-sm-6 col-6"
      >
        <h1
          class="fw-bold m-0"
        >
          GLOBAL
        </h1>
        <p>
           Total Cases
        </p>
      </div>
    </div>
    <div
      class="pt-3 m-0 mt-4 bg-pink-dark row"
      style="cursor: pointer;"
    >
      <div
        class="row"
      >
        <div
          class="col-md-2 col-sm-12 col-12"
        >
          <h6
            class="text-white fw-bold m-0 p-2 d-inline-block"
          >
            STATS BY COUNTRY
          </h6>
        </div>
        <div
          class="col-md-4 col-sm-12 col-12"
        >
          <div
            class="mb-3 input-group"
          >
            <span
              class="input-group-prepend"
            />
            <input
              class="form-control no-shadow"
              placeholder="Search Specific Country"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
